drivers: serial: Rename lpc usart shim driver

MaureenHelm opened the pull request:
Renames the lpc usart shim driver to more accurately reflect the
flexcomm hardware IP and to prepare for instantiating it on an SoC
outside the LPC family.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

Depends on zephyrproject-rtos/hal_nxp#23

**Update (@ioannisg ): seems to me the dependence is now satisfied.**

MaureenHelm committed:
drivers: serial: Rename lpc usart shim driver

Renames the lpc usart shim driver to more accurately reflect the
flexcomm hardware IP and to prepare for instantiating it on an SoC
outside the LPC family.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

agansari commented on drivers/serial/uart_mcux_flexcomm.c:
```diff
@@ -262,59 +262,59 @@ static int usart_mcux_lpc_init(struct device *dev)
+static const struct mcux_flexcomm_config mcux_flexcomm_0_config = {
 	.base = (USART_Type *)DT_USART_MCUX_LPC_0_BASE_ADDRESS,
 	.clock_source = 0,
 	.baud_rate = DT_USART_MCUX_LPC_0_BAUD_RATE,
```
I would replace these lines with the following:
```
	.base = (USART_Type *)DT_NXP_LPC_USART_USART_0_BASE_ADDRESS,
	.clock_source = kCLOCK_FlexComm0,
	.baud_rate = DT_NXP_LPC_USART_USART_0_CURRENT_SPEED,
```
Or their equivalent Flexcomm name and drop USART defines in fixup.

MaureenHelm replied:
Done

MaureenHelm committed:
drivers: serial: Use generated dts macros in mcux flexcomm driver

Uses the generated device tree macros, DT_NXP_LPC_USART_USART_*, in the
mcux flexcomm driver and removes the now unused dts fixups from the
lpc54xxx and lpc55xxx socs.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>
